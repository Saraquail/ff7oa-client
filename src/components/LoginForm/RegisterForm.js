import React, { Component } from 'react'
import AuthApiService from '../../services/auth-api-service'
import TokenService from '../../services/token-service'
import { withRouter } from 'react-router-dom';


class RegisterForm extends Component {

  handleSubmitRegisterJWT = ev => {
    ev.preventDefault()
    const { user_name, password } = ev.target

    AuthApiService.postUser({
      user_name: user_name.value,
      password: password.value
    })
      .then(user => { 
        debugger
        this.onRegistrationSuccess(user_name.value, password.value)
        

        //need a function that sends user to bestiary
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  onRegistrationSuccess = (user_name, password) => {
    AuthApiService.postLogin({
      user_name: user_name,
      password: password
    })
      .then(res => {
        user_name.value = ''
        password.value = ''
        TokenService.saveAuthToken(res.authToken)
        this.props.history.push('/bestiary')

      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render () {

    return (
      <form 
        className="register-form"
        onSubmit={this.handleSubmitRegisterJWT}
      >
        <label htmlFor="user_name">Username</label>
          <input type="text" name="user_name" id="user_name" required="" defaultValue="guest" />
        <label htmlFor="password">Password</label>
          <input type="password" name="password" id="password" required="" defaultValue="guest" />
        <button 
          type="submit">
            Let's Mosey
        </button>
      </form>
    )
  }
}

export default withRouter(RegisterForm)