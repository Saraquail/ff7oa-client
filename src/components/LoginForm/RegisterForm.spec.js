import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer';

import RegisterForm from './RegisterForm.js'

describe('<RegisterForm/>', () => {

  it('renders without crashing', () => {
  const div = document.createElement('div')

  ReactDOM.render(
  <BrowserRouter>
  <RegisterForm />
  </BrowserRouter>, div)

  ReactDOM.unmountComponentAtNode(div)
  })

  it('renders the UI as expected', () => {
    const tree = renderer
  .create(
    <BrowserRouter>
      <RegisterForm/>
    </BrowserRouter>
  )
  .toJSON();
    expect(tree).toMatchSnapshot();  
  })
})
