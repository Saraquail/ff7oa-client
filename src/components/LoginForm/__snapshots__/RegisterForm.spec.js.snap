// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RegisterForm/> renders the UI as expected 1`] = `
<div
  className="register-container"
>
  <h2
    id="register-header"
  >
    Register a new account:
  </h2>
  <form
    className="login-form"
    name="login-form"
    onSubmit={[Function]}
  >
     
    <p
      className="password-instructions onboarding"
    >
      Your password must be longer than 6 characters, and include at least one of the following: lowercase letter, uppercase letter number. 
    </p>
    <label
      htmlFor="user_name"
    >
      Username
    </label>
    <input
      id="user_name"
      name="user_name"
      required=""
      type="text"
    />
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      onChange={[Function]}
      required=""
      type="password"
    />
    
    <button
      className="lets-mosey"
      type="submit"
    >
      Let's Mosey
    </button>
  </form>
</div>
`;
